packages/logging-bunyan/src/index.js

```diff
@@ -131,6 +131,26 @@ LoggingBunyan.prototype.write = function(record) {
     );
   }
 
+  // Stackdriver Logs Viewer picks up the summary line from the 'message' field
+  // of the payload.
+  // https://cloud.google.com/logging/docs/view/logs_viewer_v2#expanding
+  //
+  // Furthermore, for error messages at severity 'error' and higher, Stackdriver
+  // Error Reporting will pick up error messages if the full stack trace is
+  // included in the textPayload or in the `message` property of the
+  // jsonPaylod. See:
+  // https://cloud.google.com/error-reporting/docs/formatting-error-messages
+  //
+  // We only do this if the user hasn't provided a message property themselves.
+  //
+  if (!record.message) {
+    if (record.err && record.err.stack) {
+      record.message = record.err.stack;
```

Comment:
I have reduced the scope of this PR to just better formatting for the Log Viewer. Error Reporting is no longer in scope.

Yes, the Error Reporting service does seem to require a `serviceContext` to properly pick up errors. But given the large conceptual overlap between Logging's `resource` and `serviceContext` I do not think it is a good idea for us to expose both of these documents to the users of these libraries – not to mention the extra bytes on the wire to send both these pieces of information on log entries.

I am working with the Logging/Error Reporting  to get some clarity on why both of these needed.